proposals/lip-0057.md

```diff
@@ -1840,7 +1852,6 @@ if the chain is the mainchain and b is the snapshot block for the migration:
             validator.address,
             validator.generatorKey
         )
-        updateDelegateEligibility(validator.address, 0)
```

Comment:
This is removed as we have the implicit assumption that all delegates without BLS key satisfy `isBanned == True`? To me it seems cleaner to not make any assumptions about BLS keys within DPoS and always initialize the eligibility for all delegates.

Reply:
The reason to remove this is that we know that `updateDelegateEligibility` will not include them in the eligible delegates substore, so by having this line, you perform useless checks for all delegates registered in mainchain..

Moreover, I don't see how you avoid to make assumptions about BLS keys in DPoS, since we already have this "if" condition, whose only reason of existence is exactly the corner case of mainchain validators with no BLS key.

Reply:
Okay, sounds good to remove it here as this logic is anyways for the Lisk Mainchain only.